### PR TITLE
permission-based jail functionality, if server owners want to have a say in what their players can do in jail

### PR DESCRIPTION
**Pull request by ghost**
adds permissions that allows server owner to choose what they want jailed players to be able to do, this includes general interaction, placing/breaking of blocks, as well as damage.

adds the functionality requests in https://github.com/EssentialsX/Essentials/issues/2336

if maintainers prefer, I can update this to be permission-based over configuration based, as that would allow these features to be adjusted to certain players.

### CODE REVIEW

**Comment by ghost** [comment]
in fact, I went ahead and implemented permissions over configuration since I think that'd it be way easier to manage and restrict / take away based on players.

allow breaking blocks: `essentials.jail.allow-break`
allow placing blocks: `essentials.jail.allow-place`
allow block damage: `essentials.jail.allow-block-damage`
allow interact: `essentials.jail.allow-interact`

File: Essentials/src/com/earth2me/essentials/Jails.java [hunk 1]
```diff
@@ -160,23 +160,23 @@ public int getCount() {
         @EventHandler(priority = EventPriority.LOW, ignoreCancelled = true)
         public void onJailBlockBreak(final BlockBreakEvent event) {
             final User user = ess.getUser(event.getPlayer());
-            if (user.isJailed()) {
+            if (user.isJailed() && !event.getPlayer().hasPermission("essentials.jail.allow-break")) {
```
Comment (mdcfe) [inline Essentials/src/com/earth2me/essentials/Jails.java]: Use Essentials' `user.isAuthorized` for permissions lookups - this uses EssentialsX's perms handling code.

File: Essentials/src/config.yml [hunk 2]
```diff
@@ -811,6 +811,7 @@ protect:
       thunder: false
       lightning: false
 
+
```
Comment (mdcfe) [inline Essentials/src/config.yml]: Any chance you could remove the extra lines that were added in this file?

**Review by mdcfe — CHANGES_REQUESTED** [review]

File: Essentials/src/config.yml [hunk 3]
```diff
@@ -907,4 +907,8 @@ spawn-on-join: false
 #- guests
 #- admin
 
+
```
Comment (Ichbinjoe) [inline Essentials/src/config.yml:910]: ?

File: Essentials/src/messages.properties [hunk 4]
```diff
@@ -141,6 +141,7 @@ gcmax=\u00a76Maximum memory\:\u00a7c {0} MB.
 gctotal=\u00a76Allocated memory\:\u00a7c {0} MB.
 gcWorld=\u00a76{0} "\u00a7c{1}\u00a76"\: \u00a7c{2}\u00a76 chunks, \u00a7c{3}\u00a76 entities, \u00a7c{4}\u00a76 tiles.
 geoipJoinFormat=\u00a76Player \u00a7c{0} \u00a76comes from \u00a7c{1}\u00a76.
+geoipCantFind=\u00a76Player \u00a7c{0} \u00a76comes from \u00a7aan unknown country\u00a76.
```
Comment (Ichbinjoe) [inline Essentials/src/messages.properties:144]: This commit overlaps with another pull request.

**Review by Ichbinjoe — CHANGES_REQUESTED** [review]

**Pull request closed without merging**